interview-questions.md

```diff
@@ -9,35 +9,66 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
 Your answer:
-
+The name of the foreign key would be cohort_id, on the student model.  We can fix the mistake by adding columns then running a migration. 
 Researched answer:
+We can fix the mistake by opening Terminal and running $ rails generate migration add_columns_to_list.  We should see a file in the migrate folder.  If we select that migration file, we can add a column with - add_column :students, :integer, :cohort_id (cohort_id is the name of the foreign key.  It will be on the Student model because the students will need to be associated with the appropriate cohort).  Return to Terminal and run $ rails db:migrate.
 
 2. Which RESTful routes must always be passed params? Why?
 
 Your answer:
-
+The RESTful routes that must be passed through params are Show (displaying a specific instance), Create (strong params require allowable data), Edit (editing the desired instance), Update (we need strong params to verify permissible data), and Destroy (pass through params for the desired instance).
 Researched answer:
+Show - In order to see the desired instance, we must issue a GET command with specified params.
+Create - Using a POST request, Create uses strong params to issue specific requirements for the data to be considered valid and thereby, useable.  Without them, data submitted for instance creation could fall outside of intended program usage.
+Edit - In order to edit a specific instance, we need to use a GET request and pass the appropriate params and id.
+Update - We can update an instance with the PATCH/PUT request.  Strong params are needed in order to make sure only viable data is being utilized.
+Destroy - A DELETE request, params allow for a specific instance to be removed.
```

Comment:
Awesome! We definitely need to identify the data we are manipulating which makes sense why it is required for these restful routes.